### PR TITLE
code deduplication: image filename validation

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

Refactoring around `yt.funcs.get_image_suffix`, which gives noisy warnings on Travis such as

```
WARNING:yt:Unsupported image suffix requested (.h5)
WARNING:yt:Unsupported image suffix requested (.h5_Slice_z_density)
WARNING:yt:Unsupported image suffix requested (.h5)
WARNING:yt:Unsupported image suffix requested (.h5_Projection_z_density)
```
while the images are still saved in the end, as png.
Since calls to this function are almost _always_ followed by some way to default to `".png"` as a suffix, I
replaced this function with `yt.visualization._commons.validate_image_name`
Doing so, I ended up removing duplicated code with the function `get_canvas`, which already existed but wasn't used

Since some of the affected parts of the code use a lot of nested ifs, I'm pretty sure I must have broken stuff on the way there, so I'm opening as a draft for now🤞 

## PR Checklist

- [x] pass `black --check yt/`
- [x] pass `isort . --check --diff`
- [x] pass `flake8 yt/`
- [x] pass `flynt yt/ --fail-on-change --dry-run -e yt/extern`
- [ ] New features are documented, with docstrings and narrative docs
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

**Comment by neutrinoceros** [comment]
Hum. Once again I broke some answer tests and I'm having a hard time navigating the source code to find out what's going wrong (or specifically, which part of my change is causing this). @Xarthisius, can I ask your help with this ?

**Comment by neutrinoceros** [comment]
Nevermind I found where it happened by encapsulating the error.

File: yt/visualization/base_plot_types.py
Comment (neutrinoceros) [inline yt/visualization/base_plot_types.py:17]: nothing is currently achieved by this change. I was trying to look if this dictionary could be useful elsewhere, and refactored a bit around it to make sense out of it. Now I don't think this would actually be useful as part of yt/visualization/_commons.py but the code that uses it is now slightly more readable, so I think I'll just keep it as is.

File: yt/visualization/profile_plotter.py
Comment (neutrinoceros) [inline yt/visualization/profile_plotter.py:315]: I note that I broke that part of the logic tree. I think this is one of the places where we implement a "smart" pattern to allow a full path name (that is, the exact desired name of the output file, with an extension included) be passed, while this is not the general way output file names are constructed.
Meanwhile, it doesn't make much tore use a unique exact name inside a loop as is done here.
All in all, I think the code would be much more maintainable if we renounced inferring an extension from the `name` argument, _at least_ when more than one image file is to be saved.

File: yt/visualization/profile_plotter.py
Comment (neutrinoceros) [inline yt/visualization/profile_plotter.py:1277]: see comment above

File: yt/visualization/plot_container.py
Comment (neutrinoceros) [inline yt/visualization/plot_container.py:499]: see below

**Comment by neutrinoceros** [comment]
@yt-fido test this please

**Comment by neutrinoceros** [comment]
@yt-fido test this please

**Review by matthewturk — APPROVED** [review]

**Comment by matthewturk** [comment]
@neutrinoceros You want any support on this?

**Comment by neutrinoceros** [comment]
I think it's done ? Just waiting for a second review afaic :-)

**Comment by neutrinoceros** [comment]
note that this would benefit from #3064 regarding type hints

**Review by munkm — APPROVED** [review]

**Pull request merged**
